File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/Ubicacion.java
package com.miapp.mi_servidor.Enums;

/**
 *
 * @author Kathy
 */
public enum Ubicacion {
    AZUAY("Azuay"),
    BOLIVAR("Bolivar"),
    CAÑAR("Cañar"),
    CARCHI("Carchi"),
    CHIMBORAZO("Chimborazo"),
    COTOPAXI("Cotopaxi"),
    EL_ORO("El Oro"),
    ESMERALDAS("Esmeraldas"),
    GALAPAGOS("Galapagos"),
    GUAYAS("Guayas"),
    IMBABURA("Imbabura"),
    LOJA("Loja"),
    LOS_RIOS("Los Rios"),
    MANABI("Manabi"),
    MORONA_SANTIAGO("Morona Santiago"),
    NAPO("Napo"),
    ORELLANA("Orellana"),
    PASTAZA("Pastaza"),
    PICHINCHA("Pichincha"),
    SANTA_ELENA("Santa Elena"),
    SANTO_DOMINGO_DE_LOS_TSACHILAS("Santo Domingo de los Tsachilas"),
    SUCUMBIOS("Sucumbios"),
    TUNGURAHUA("Tungurahua"),
    ZAMORA_CHINCHIPE("Zamora Chinchipe");

    private final String displayName;

    Ubicacion(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

     // Método fromValue
     public static Ubicacion fromValue(String value) {
        for (Ubicacion ubicacion : Ubicacion.values()) {
            if (ubicacion.getDisplayName().equalsIgnoreCase(value)) {
                return ubicacion;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum Ubicacion: " + value);
    }
}

File: Proyecto LP/mi-servidor/Frontend/lib/Enums/Ubicacion.dart
enum Ubicacion {
  azuay("Azuay", "AZUAY"),
  bolivar("Bolivar", "BOLIVAR"),
  canar("Cañar", "CAÑAR"),
  carchi("Carchi", "CARCHI"),
  chimborazo("Chimborazo", "CHIMBORAZO"),
  cotopaxi("Cotopaxi", "COTOPAXI"),
  elOro("El Oro", "EL_ORO"),
  esmeraldas("Esmeraldas", "ESMERALDAS"),
  galapagos("Galapagos", "GALAPAGOS"),
  guayas("Guayas", "GUAYAS"),
  imbabura("Imbabura", "IMBABURA"),
  loja("Loja", "LOJA"),
  losRios("Los Rios", "LOS_RIOS"),
  manabi("Manabi", "MANABI"),
  moronaSantiago("Morona Santiago", "MORONA_SANTIAGO"), 
  napo("Napo", "NAPO"),
  orellana("Orellana", "ORELLANA"),
  pastaza("Pastaza", "PASTAZA"),
  pichincha("Pichincha", "PICHINCHA"),
  santaElena("Santa Elena", "SANTA_ELENA"),
  santoDomingoDeLosTsachilas("Santo Domingo de los Tsachilas", "SANTO_DOMINGO_DE_LOS_TSACHILAS"),
  sucumbios("Sucumbios", "SUCUMBIOS"),
  tungurahua("Tungurahua", "TUNGURAHUA"),
  zamoraChinchipe("Zamora Chinchipe", "ZAMORA_CHINCHIPE");

  final String displayName;
  final String backendValue;

  const Ubicacion(this.displayName, this.backendValue);

  static String? getBackendValue(String? displayName) {
    return Ubicacion.values.firstWhere(
      (ubicacion) => ubicacion.displayName == displayName,
      orElse: () => throw Exception("Valor inválido para el enum Ubicacion: $displayName"),
    ).backendValue;
  }
}